iceoryx_hoofs/include/iceoryx_hoofs/cxx/helplets.hpp

```diff
@@ -351,9 +351,33 @@ constexpr T from(const F value) noexcept;
 /// @param[in] value of type F to convert to T
 /// @return converted value of F to corresponding value of T
 template <typename T, typename F>
-constexpr T into(const F value) noexcept;
-} // namespace cxx
-} // namespace iox
+constexpr T into(const F value) noexcept {
+Into<F, T>::into(value); // just forwarding to the struct
+}
+
+template<uint64_t N>
```

Comment:
@NNatarajan0717 Thanks for your contribution! I would refactor this could a bit.

1. First of all we have to declare a struct which can be implemented anywhere in the code base so that users are able to define custom conversions and use `into`. I named it `IntoImpl` and when a user defines it for custom `Source` and `Destination` then one can use `into` with it.
```cxx
template <typename Destination, typename Source>
struct IntoImpl;
```

2. Let's define a partial struct specialization for `std::string` from `cxx::string` with any arbitrary `Capacity`
```cxx
template <uint64_t Capacity>
struct IntoImpl<std::string, iox::cxx::string<Capacity>>
{
    static std::string into(const iox::cxx::string<Capacity>& source) noexcept
    {
        return std::string{source.c_str()};
    }
};
```

3. Another example could be the conversion from integer into `cxx::string` for any arbitrary `Capacity`
```cxx
template <uint64_t Capacity>
struct IntoImpl<iox::cxx::string<Capacity>, int>
{
    static iox::cxx::string<Capacity> into(const int source) noexcept
    {
        constexpr uint64_t BUFFER_CAPACITY = 11; // see ceil(log_10(sizeof(int))) + 1 (for null terminator)
        static_assert(Capacity >= BUFFER_CAPACITY, "String capacity is too small to contain integer");

        char buffer[BUFFER_CAPACITY];
        int length = snprintf(&buffer[0], BUFFER_CAPACITY, "%d", source); s: maxlen: format:
        return iox::cxx::string<Capacity>(iox::cxx::TruncateToCapacity, &buffer[0], static_cast<uint64_t>(length));
    }
};
```

4. Finally, it is also possible to implement a plain old conversion from `int` to `float` - just as an example, not required since we have `static_cast`.
```cxx
template <>
struct IntoImpl<float, int>
{
    static float into(const int source)
    {
        return static_cast<float>(source);
    }
};
```

5. With the declaration of the unimplemented `IntoImpl` alone we can implement an `into` function which converts the types if an implementation exists. If not the compiler prints out a compile time error message.
```cxx
template <typename Destination, typename Source>
Destination into(const Source& source) noexcept
{
    return IntoImpl<Destination, Source>::into(source);
}
```

So when some user would like to use it they can simply write:
```cxx
 auto ioxString = into<iox::cxx::string<100>>(5);
 auto stdString = into<std::string>(ioxString); :
```

With this in mind could you please adjust the code that only a single `into` function exists. Some more additional hints, since we are working in a safety critical environment we:
* are not allowed to use heap or exceptions therefore I used `snprintf` for conversion
* we have to follow our `.clang-tidy` ruleset strictly, so please active `clang-tidy` in your IDE and fix all warnings you may encounter
* We require tests for all new constructs. The tests for the helplets you can find `./iceoryx_hoofs/tests/test_cxx_helplets.cpp`. Every tests requires a unique id which you can generate with the command line tool `uuidgen`. Just take a look at the other tests.

If you have further questions do not hesitate to ping me. It is a lot and it is a challenge but I would be happy to guide you trough this process and with the next PR it works then more smoothly :smile: